### PR TITLE
Cleanup formatting

### PR DESCRIPTION
**Pull request by BrunoMSantos**
Looking over Robin's #86 I had a few ideas and started poking around only to notice that the parser and docstring modules still had a few blurry lines in their boundaries. This is a quick attempt at fixing those issues so that the parser has 0 effect in the output formatting. It also makes the formatting a bit more consistent overall.

This will conflict with #86 though it's not a big issue and I have local patches for it already if it matters.

### CODE REVIEW

File: hawkmoth/docstring.py [hunk 1]
```diff
@@ -142,7 +142,14 @@ def get_docstring(self, transform=None):
 
         text = Docstring._nest(text, self._indent)
 
-        args = ', '.join(self._args) if self._args is not None else None
+        name = self._get_decl_name()
```
Comment (jnikula) [inline hawkmoth/docstring.py:145]: I guess you meant to use name later, but didn't?

File: hawkmoth/docstring.py [hunk 2]
```diff
@@ -142,7 +142,14 @@ def get_docstring(self, transform=None):
...
+
+        ttype = self._ttype
+
+        args = ''
```
Comment (jnikula) [inline hawkmoth/docstring.py:153]: Feels like this should remain None if there are no args. If it "leaks" to `_fmt.format()` I think it'll be more obvious there's a bug somewhere?

Comment (jnikula) [reply hawkmoth/docstring.py:153]: Ahah, used in the 2nd commit.

**Review by jnikula — COMMENTED** [review]
Overall I like it. The whitespace stuff has annoyed me too. And I agree with the spacing `int *foo` over anything else.

Couple of nits, I could lean either way with them.

File: hawkmoth/docstring.py (hunk 2 above)
Comment (jnikula) [reply hawkmoth/docstring.py:153]: Sheesh github landed this comment in a silly place. This was related to `name = self._get_decl_name()`.

File: hawkmoth/docstring.py (hunk 1 above)
Comment (BrunoMSantos) [reply hawkmoth/docstring.py:145]: Ah, it was for consistency and readability, I wanted to have a smaller name to pass to the function. But when I decided the commits made more sense in the reverse order, I clearly messed up this part of the rebase.

I'll probably need to force push to resolve it.

File: hawkmoth/docstring.py (hunk 2 above)
Comment (BrunoMSantos) [reply hawkmoth/docstring.py:153]: So I may be misreading this sequence of comments, but I'm guessing the 1st comment is still about `args = ''`? If so, note that `self._args` may still be `None`, `args` on the other hand is the processed argument string we want to use in the final line and that has to be an actual string. Did I misunderstand anything?

By the way, there could be a point in handling `self._args` in the derived classes. I could create a method to abstract the argument stitching. Same for the type spacer thing. It felt a bit verbose in comparison, but might make it nicer overall.

Comment (jnikula) [reply hawkmoth/docstring.py:153]: I meant that it's a (theoretical) functional change to set `args = ''`.  It was set to `None` before, but it's fine when `self._fmt` does not contain `{args}`.
Anyway, not a big deal, I think I'll just merge this as-is.
Thanks for doing this!

Comment (BrunoMSantos) [reply hawkmoth/docstring.py:153]: Ah, I see now. Well, I didn't fix the `name` and `ttype` being defined one commit ahead, but alright.
Thanks for the review!

**Pull request merged**